terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -157,7 +157,8 @@ def registry_declaration_regex
         %r{
           (?<=\{)
           (?:(?!^\}).)*
-          source\s*=\s*["'](#{Regexp.escape(registry_host_for(dependency))}/)?#{Regexp.escape(dependency.name)}["']
+          source\s*=\s*["'](#{Regexp.escape(registry_host_for(dependency))}/)?
+            #{Regexp.escape(dependency.name)}(//modules/[^"']+)?["']
```

Comment:
> (//modules/[^"']+)?

A couple of questions:

* Is the `^` needed here?
* I was expecting to see `\/` to escape the slash but I think that `%r` removes the need for escaping. Is the extra leading `/` needed?
* What is the optional `[^"'` mean to handle? Examples if possible pretty please.

Is this the equivalent of?

```ruby
          source\s*=\s*["']
            (#{Regexp.escape(registry_host_for(dependency))}/)?
            #{Regexp.escape(dependency.name)}
            (/modules/\S*)?
            ["']
```

Reply:
The optional part is attempting to match the nested module path.
Input: `source  = "terraform-aws-modules/iam/aws//modules/iam-assumable-role"` match `//modules/iam-assumable-role`, terraform enforces the `//` in the path.

I was attempting to match everything after `//modules/` up until the the ending quote with `[^'"]+` but `\S+` would be equivalent 👍